lib/puppet/functions/simplib/rand_cron.rb

```diff
@@ -0,0 +1,144 @@
+  #   basis for the returned values.  This algorithm works well to
+  #   create cron job intervals for multiple hosts, when the number
+  #   of hosts is less than the `max_value` or the hosts do not have
+  #   linearly-assigned IP addresses.
+  #
+  #   When 'ip_mod' and the input string is an IP address, the modulus
+  #   of the numeric IP is used as the basis for the returned values.
+  #   This algorithm works well to create cron job intervals for
+  #   multiple hosts, when the number of hosts exceeds the `max_value`
+  #   and the hosts have linearly-assigned IP addresses.
+  #
+  #   When 'ip_mod' and the input string is not an IP address, for
+  #   backward compatibility,  the crc32 of the input string will
+  #   be used as the basis for the returned values.
+  #
+  #   When 'crc32', the crc32 of the input string will be used as the
+  #   basis for the returned values.
+  #
+  # @param occurs
+  #   The occurrence within an interval, i.e., the number of values to
+  #   be generated for the interval. Defaults to `1`.
+  #
+  # @param max_value
+  #   The maximum value for the interval.  The values generated will
+  #   be in the inclusive range [0, max_value]. Defaults to `60` for
+  #   use in the `minute` cron field.
+  #
+  # @return [Array[Integer]] Array of integers suitable for use in the
+  #   ``minute`` or ``hour`` cron field.
+  #
+  # @example Generate one value for the `minute` cron interval using
+  #   the 'sha256' algorithm
+  #
+  #   rand_cron('myhost.test.local','sha256')
+  #
+  # @example Generate 2 values for the `minute` cron interval using
+  #   the 'sha256' algorithm applied to the numeric representation of
+  #   an IP
+  #
+  #   rand_cron('10.0.23.45', 'sha256')
+  #
+  # @example Generate 2 values for the `hour` cron interval, using the
+  #   'ip_mod' algorithm
+  #
+  #   rand_cron('10.0.6.78', 'ip_mod', 2, 23)
+  #
+  dispatch :rand_cron do
+    required_param 'String',            :modifier
+    required_param 'RandCronAlgorithm', :algorithm
+    optional_param 'Integer[1]',        :occurs
+    optional_param 'Integer[1]',        :max_value
+  end
+
+  def generate_crc32_number(input_string)
+    require 'zlib'
+    Zlib.crc32(input_string)
+  end
+
+  def generate_ip_mod_number(input_string)
+    require 'ipaddr'
+
+    ip_num = nil
+    begin
+     ip_num = IPAddr.new(input_string).to_i
+    rescue IPAddr::Error
+    end
+
+    num = nil
+    if ip_num.nil?
+      # crc32 calculation for backward compatibility
+      num = generate_crc32_number(input_string)
+    else
+      num = ip_num
+    end
+    num
+  end
+
+  def generate_sha256_number(input_string)
+    require 'ipaddr'
+    require 'digest'
+
+    ip_num = nil
+    begin
+     ip_num = IPAddr.new(input_string).to_i
+    rescue IPAddr::Error
+    end
+
+    num = nil
+    if ip_num.nil?
+      num = Digest::SHA256.hexdigest(input_string).hex
+    else
+      num = Digest::SHA256.hexdigest(ip_num.to_s).hex
```

Comment:
Is it necessary to convert the ip to an integer, then back to a string before calling SHA256? the hashing algorithm should provide the same statistical distribution from "192.168.251.1" to '12351123516', between numbers.